### PR TITLE
Fix: add validation set to training function

### LINKED ISSUE #20: add validation set
We need to add a validation set. We have at least two options.

1. split training set into train/validation sets. there can be a user-defined parameter for the split ratio and a random seed to control 
2. add a validation set in the hdf5 file

we will go with option 1 in the short term because it will take less time and effort. but it would be nice to implement 2.

### PR DESCRIPTION
**Pull request by kaczmarj**
This PR creates a validation set from the training set. This means we would train on training set, use validation set for early stopping, and then test on the test set. The test set should not be used until the model is ready to be evaluated after all training has finished.

Fixes #20

### CODE REVIEW

**Comment by kaczmarj** [comment]
@jhajagos - how does this look to you?
